features/FEATURE_LWIP/TESTS/mbedmicro-net/udp_echo_parallel/main.cpp

```diff
@@ -68,70 +86,95 @@ class Echo {
 
         sock.set_timeout(MBED_CFG_UDP_CLIENT_ECHO_TIMEOUT);
 
-        for (int i = 0; i < ECHO_LOOPS; i++) {
-            prep_buffer(tx_buffer, sizeof(tx_buffer));
+        int i = 0;
+        while(success < ECHO_LOOPS) {
+            prep_buffer(id, uuid_buffer, uuid_len, tx_buffer, sizeof(tx_buffer));
             const int ret = sock.sendto(udp_addr, tx_buffer, sizeof(tx_buffer));
             iomutex.lock();
-            printf("[%02d] sent...%d Bytes \n", i, ret);
+            printf("[ID:%01d][%02d] sent %d Bytes - %.*s\n", id, i, ret, MBED_CFG_UDP_CLIENT_ECHO_BUFFER_SIZE, tx_buffer);
             iomutex.unlock();
 
             SocketAddress temp_addr;
             const int n = sock.recvfrom(&temp_addr, rx_buffer, sizeof(rx_buffer));
             iomutex.lock();
-            printf("[%02d] recv...%d Bytes \n", i, n);
+            printf("[ID:%01d][%02d] recv %d Bytes - %.*s\n", id, i, ret, MBED_CFG_UDP_CLIENT_ECHO_BUFFER_SIZE, rx_buffer);
             iomutex.unlock();
 
             if ((temp_addr == udp_addr &&
                  n == sizeof(tx_buffer) &&
                  memcmp(rx_buffer, tx_buffer, sizeof(rx_buffer)) == 0)) {
                 success += 1;
+                iomutex.lock();
+                printf("[ID:%01d][%02d] success #%d\n", id, i, success);
+                iomutex.unlock();
             }
+
+            i++;
         }
 
+        result = success == ECHO_LOOPS;
+
         err = sock.close();
         TEST_ASSERT_EQUAL(0, err);
```

Comment:
I don't believe so, we want to know what the error code is so an assert seems like the right choice here.